typegate/tests/utils/tg_deploy_script.py

```diff
@@ -0,0 +1,74 @@
+import importlib.util as import_util
+import os
+import sys
+import json
+
+from typegraph.gen.exports.core import (
+    ArtifactResolutionConfig,
+    MigrationAction,
+    MigrationConfig,
+)
+from typegraph.graph.shared_types import BasicAuth
+from typegraph.graph.tg_deploy import TypegraphDeployParams, tg_deploy
+
+# get command args
+cwd = sys.argv[1]
+PORT = sys.argv[2]
+module_path = sys.argv[3]
+secrets_str = sys.argv[4]
+
+gate = f"http://localhost:{PORT}"
+auth = BasicAuth("admin", "password")
+
+# resolve the module
+module_name = os.path.basename(module_path)
+spec = import_util.spec_from_file_location(module_name, module_path)
+module = import_util.module_from_spec(spec)
+spec.loader.exec_module(module)
+
+tg_name = module_name.split(".")[0]
+if not hasattr(module, tg_name):
```

Comment:
We could also check the type of the exported symbols instead... It can work with multiple typegraphs in a single file...

Reply:
I haven't thought of it this way, but a better approach for sure 🔥. I will make the change,

Reply:
@Natoandro But thinking more about it more now, which typegraph will be used to initiate the engine in the test? I haven't seen a multi typegraph test before, do we need a deploy script which will handle a file with multiple typegraphs.

Reply:
@destifo you can definitely add one 👍

Reply:
so the thing is, if we have multiple typegraphs,`t.engine` should return as many query engines as the number of typegraphs we have. We would need to route queries to the correct query engine. This would require us to change `t.engine`, but the downside of this is that it will make `t.engine` bloated, handling multiple logics. For multiple typegraphs, I was thinking of adding a new method `t.engines` which will handle a file with multiple typegraphs. wdyt @zifeo @Natoandro